src/main/java/frc/robot/subsystems/Leds.java

```diff
@@ -0,0 +1,251 @@
+  private static final double autoFadeMaxTime = 5.0; // Return to normal
+
+
+  public Leds() {
+    leds = new AddressableLED(4);
+    buffer = new AddressableLEDBuffer(length);
+    leds.setLength(length);
+    leds.setData(buffer);
+    leds.start();
+    loadingNotifier =
+      new Notifier(
+        () -> {
+          synchronized (this) {
+            breath(Color.kWhite, Color.kBlack, System.currentTimeMillis() / 1000.0);
+            leds.setData(buffer);
+          }
+        });
+    loadingNotifier.startPeriodic(0.02);
+  }
+
+  public synchronized void periodic() {
+    if (DriverStation.isFMSAttached()) {
+      alliance = DriverStation.getAlliance();
+      allianceColor = 
+        alliance
+          .map(alliance -> alliance == Alliance.Blue ? Color.kBlue : Color.kRed)
+          .orElse(Color.kBurlywood);
+      secondaryDisabledColor = alliance.isPresent() ? Color.kBlack : Color.kThistle;
+    }
+
+    // Update auto state
+    if (DriverStation.isDisabled()) {
+      autoFinished = false;
+    } else {
+      lastEnabledAuto = DriverStation.isAutonomous();
+      lastEnabledTime = Timer.getFPGATimestamp();
+    }
+
+    // Update estop state
+    if (DriverStation.isEStopped()) {
+      estopped = true;
+    }
+
+    // Exit during initial cycles
+    loopCycleCount += 1;
+    if (loopCycleCount < minLoopCycleCount) {
+      return;
+    }
+
+    // Stop loading notifier if running
+    loadingNotifier.stop();
+
+    // Select LED mode
+    solid(Color.kBlack); // Default off
+    if (estopped) {
+      solid(Color.kRed);
+    } else if (DriverStation.isDisabled()) {
+      if (lastEnabledAuto && Timer.getFPGATimestamp() - lastEnabledTime < autoFadeMaxTime) {
+        // Auto fade
+        solid(1.0 - ((Timer.getFPGATimestamp() - lastEnabledTime) / autoFadeTime), Color.kGreen);
+       } else if (prideLeds) {
+        // Pride stripes
+        stripes(
+            List.of(
+                Color.kBlack,
+                Color.kRed,
+                Color.kOrangeRed,
+                Color.kYellow,
+                Color.kGreen,
+                Color.kBlue,
+                Color.kPurple,
+                Color.kBlack,
+                new Color(0.15, 0.3, 1.0),
+                Color.kDeepPink,
+                Color.kWhite,
+                Color.kDeepPink,
+                new Color(0.15, 0.3, 1.0)),
+            50,
+            5.0);
+        buffer.setLED(staticLength, allianceColor);
+      } else {
+        // Default pattern
+        wave(allianceColor, secondaryDisabledColor, waveAllianceCycleLength, waveAllianceDuration);
+      }
+    }
+      if (DriverStation.isAutonomous()) {
+        wave(Color.kDarkGreen, Color.kPurple, waveSlowCycleLength, waveSlowDuration);
+        if (autoFinished) {
+          double fullTime = (double) length / waveFastCycleLength * waveFastDuration;
+          solid((Timer.getFPGATimestamp() - autoFinishedTime) / fullTime, Color.kGreen);
+        }
+      } else { // Enabled
```

Comment:
The existing colors for intaking should be maintained to reduce driver confusion

Reply:
panav said these are fine plus this is what he wanted originally so